client/transport.c

```diff
@@ -67,19 +70,21 @@ static int process_transport_block(struct xdag_block *received_block, void *conn
 	xtime_t start_time = xdag_start_main_time();
 	xtime_t current_time = xdag_main_time();
 
-	if(current_time < start_time || stats->total_nmain > current_time - start_time + 2) return -1;
+	if(current_time >= start_time && stats->total_nmain <= current_time - start_time + 2) {
+		if(stats->main_time <= current_time + 2) {
```

Comment:
Here we suppose the latency of the network is 2 seconds?

Reply:
yes, i left it because it was already 2 seconds, we can test and reduce it